internal/collector/sync_quota_to_backend.go

```diff
@@ -190,14 +196,14 @@ func (c *Collector) performQuotaSync(ctx context.Context, srv db.ProjectService,
 			}
 		} else {
 			// for anything except AZSeparatedTopology, the total target quota is just what we have in project_resources.quota
-			if targetQuotaPtr == nil {
+			if targetQuotaPtr.IsNone() {
 				return fmt.Errorf("found unexpected NULL value in project_resources.quota for id = %d", resourceID)
 			}
-			targetQuota = *targetQuotaPtr
+			targetQuota, _ = targetQuotaPtr.Unpack()
 		}
 
 		targetQuotasInDB[resourceName] = targetQuota
-		if currentQuota == nil || *currentQuota < 0 || uint64(*currentQuota) != targetQuota {
+		if val, isSome := currentQuotaPtr.Unpack(); !isSome || val < 0 || uint64(val) != targetQuota {
```

Comment:
Please double-check my logic. But I think this can be shortened to:
```suggestion
		if currentQuotaPtr != Some(targetQuota) {
```
because Option types can be compared to each other by value (unlike pointer types).

Reply:
`if currentQuotaPtr != Some(int64(targetQuota)) {
	needsApply = true
}`
The targetQuota needs to be converted into int64 for the comparison to work. Besides from that, the logic should work. Could this cause any weird edge cases? Before we converted int64 to uint64 and now it is the other way around. What is the reason that backend_quota and quota are different types?

Reply:
>> golangci-lint
internal/collector/sync_quota_to_backend.go:207:35: G115: integer overflow conversion uint64 -> int64 (gosec)

Reply:
Argh, the conversion lint is a good point. Then we might have to keep it as-is.